openmls/src/group/core_group/new_from_external_init.rs

```diff
@@ -71,31 +66,38 @@ impl CoreGroup {
             },
         )?;
 
-        if treesync.tree_hash() != verifiable_public_group_state.tree_hash() {
+        if treesync.tree_hash() != group_info.group_context().tree_hash() {
             return Err(ExternalCommitError::TreeHashMismatch);
         }
 
-        let pgs_signer_leaf = treesync
-            .leaf(verifiable_public_group_state.signer())
-            .map_err(|_| ExternalCommitError::UnknownSender)?;
-        let pgs_signer_credential = pgs_signer_leaf
-            .ok_or(ExternalCommitError::UnknownSender)?
-            .key_package()
-            .credential();
-        let pgs: PublicGroupState = verifiable_public_group_state
-            .verify(backend, pgs_signer_credential)
-            .map_err(|_| ExternalCommitError::InvalidPublicGroupStateSignature)?;
+        // Obtain external_pub from GroupInfo extensions.
+        // TODO(#720): Check for duplicates.
+        let external_pub = {
```

Comment:
If you'd do this within `InitSecret::from_public_group_state` you wouldn't need the extra parameter for `external_pub`.

Reply:
Right. But then I need to extend `KeyScheduleError`. Is that okay? Seems to blurry that type a bit?